Lib/functools.py

```diff
@@ -753,8 +751,13 @@ def singledispatch(func):
     function acts as the default implementation, and additional
     implementations can be registered using the register() attribute of the
     generic function.
-
     """
+    # There are many programs that use functools without singledispatch, so we
+    # trade-off making singledispatch marginally slower for the benefit of
+    # making start-up of such applications slightly faster.
+    from types import MappingProxyType
```

Comment:
It is slightly faster to import `types` and use `types.MappingProxyType` once than import `MappingProxyType` from `types`. The `from ... import` case runs a Python code from `_bootstrap.py`, while the former case runs just a C code if the types module already is imported.